pkg/webhooks/customcluster_webhook__test.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright Kurator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhooks
+
+import (
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/yaml"
+
+	"kurator.dev/kurator/pkg/apis/infra/v1alpha1"
+)
+
+func TestCustomClusterValidation(t *testing.T) {
+	// read configuration from examples directory
+	caseNames := []string{"../../examples/infra/customcluster/cc-customcluster.yaml"}
+
+	wh := &CustomClusterWebhook{}
+	for _, tt := range caseNames {
+		t.Run(tt, func(t *testing.T) {
+			g := NewWithT(t)
+			c, err := readCustomCluster(tt)
+			g.Expect(err).NotTo(HaveOccurred())
+
+			err = wh.validate(c)
```

Comment:
Ther are so many invalid fields in one case, and you only checked error exists. I think it is not a complete coverage

Reply:
I understand your point about the need for complete coverage in the test cases. 
 this func `TestCustomClusterValidation` only test for the case in examples directory.

To cover potential error scenarios, I have created individual test cases in the `TestInvalidCustomClusterValidation` function